spec/services/place_hold_form/builder_spec.rb

```diff
@@ -33,6 +33,18 @@
       expect(builder.generate[:author]).to eq 'Hill Street blues (Television program)'
     end
 
+    context 'when catkey has a prefix "a"' do
+      let(:catkey) { 'a1' }
+
+      before do
+        allow(client).to receive(:get_bib_info).with('1', 'fake-token').and_return(get_bib_info_response)
```

Comment:
that catkey variable defined on 37 never gets called - shouldn't it be loaded into get_bib_info as the first argument instead of `1`? Or am I missing something?

```suggestion
        allow(client).to receive(:get_bib_info).with(catkey, 'fake-token').and_return(get_bib_info_response)
```

Reply:
it is being used by the subject on top 

`subject(:builder) { described_class.new(catkey: catkey, user_token: 'fake-token', client: client, library: 'UP-PAT') }`

since get_bib_info is called after the prefix a is deleted, needed to stub with only `1`

Reply:
ah ok cool thanks